### ISSUE TITLE
Selected video format isn't supported

### ISSUE THREAD

**Issue by GitujMe** [yuliskov/SmartTubeNext#1258]
This bug is driving me nuts! All other apps work like default YouTube, Twitch, IPTV, Kodi etc. This is not ExoPlayer bug. I know about that ExoPlayer bug where you can't watch any app which uses ExoPlayer, but in this case everything works except STN.

I was just watching an playlist and suddenly STN started saying "Selected video format isn't supported" and I can't play any video anymore. I tried every single resolution!! It doesn't make any difference if I choose avc or vp9 it occurs everywhere including Videos and Live streams. The video I successfully watched 10 minutes ago now won't play. Nothing will play. I have to switch back to default YouTube and it already displayed me like 30 ads so far it's unwatchable...

I had this problem yesterday but I was on an old version, I updated app and it worked until now when even on latest version this problem is still present.

**Comment by akzkak** [comment]
+1

Complete reinstall fixed it, but restore backup and same error spam again. Disable Sponsorblock fixed it. Likely updated something on there end and SmartTube errors out.

**Comment by GitujMe** [comment]
> +1
> 
> Complete reinstall fixed it, but restore backup and same error spam again. Disable your Sponsorblock. They likely updated something on there end and SmartTube errors out.

Yes exactly this! I actually didn't reinstalled app, I just cleared the data which is practically enough. As soon as I brought back my backup the error spam appeared again and again. Even if I'm in completely different app I still see this error spam CONSTANTLY it's so annoying.

And the worst part is that STN is actually making ExoPlayer to bug and then ExoPlayer bugs other apps too. For example TiviMate freezes when I open any channel, another IPTV app plays audio but no video.

STN is at the moment unusable and totally breaks whole device. After restarting device everything is fine until everything stops working again. I am deleting this app for good. Let me know if this gets fixed because I'm so pissed about this. I had to restart my Android TV device 6 times in past two days because of STN...

**Comment by akzkak** [comment]
Oh really bad if its effecting other apps. Disable Sponsorblock did not do anything? Still working here, not had the error again, yet.

**Comment by GitujMe** [comment]
> Oh really bad if its effecting other apps. Disable Sponsorblock did not do anything? Still working here, not had the error again, yet.

Yeah, bugged STN affected all other ExoPlayer apps like defailt YT, TiviMate and others. Simple restart fixed it.

I still didn't installed app back, is it fixed by now? :)

**Comment by DPAmes** [comment]
Getting failure to play video with this "format not supported" error message on the movie "The Beyond". SmartTubeNext only (not Youtube or Vanced), no other videos affected. First time I've seen a Youtube video that won't play in STN.

**Comment by dageekkid** [comment]
I can confirm this fails on most if not all the free ad movies on YT. This does not occur on the (Re)Vanced versions.

**Comment by jaredkotoff** [comment]
It's pretty rare but i get this on some videos.
Currently getting this on this video: https://www.youtube.com/watch?v=5lDSSgHG4q0
Any of the `vp9` formats won't play. Switching to `avc` (1080p, 30fps avc) lets it play.

SmartTube 19.59
2019 Nvidia Shield Pro 9.1.1(33.2.0.157)
Android 11
